### DISCUSSION TITLE
Clear documentation to setup non-TS project

### DISCUSSION THREAD

**Discussion by feryardiant** [vuejs/language-tools#320] [Q&A] [answered]
Hi!

Firstly, thank you for your works for creating such a great extension. Secondly, I am not sure if it was an issue of this extension or is it just lack of documentation how to setup a non-TS project. as stated in the readme:

> It's based on `@vue/reactivity` to calculate TypeScript on-demand to optimize performance similar to the native TypeScript language service.

I assume that this extension is mainly focus to bring great TypeScript support, but not really for non-TS one. Why?

## Type hover

![XTxscmujGv](https://user-images.githubusercontent.com/508665/126081001-e46b70dd-4964-470a-b96c-952008f03357.gif)

Here I just created 2 projects using `npm init @vitejs/app` command, one with `--template vue` and another one with `--template vue-ts` flag. Since it requires to have `tsconfig.json` or `jsconfig.json` in the root project, so I just copy the `tsconfig.json` from `vite-app-ts` project to `vite-app` and rename it to `jsconfig.json`. Then install the dependencies using `pnpm`

```shell
$> npm init @vitejs/app vite-app --template vue && cd vite-app && pnpm install
$> cd ..
$> npm init @vitejs/app vite-app-ts --template vue-ts && cd vite-app && pnpm install
$> cd ..
$> cp vite-app-ts/tsconfig.json vite-app/jsconfig.json
```

Take a look at `src/components/HelloWorld.vue` file from either projects and hover over the `{{ msg }}` on line 2. Notice that one on the `vite-app` shows `Cannot find name 'msg'.ts(2304)` and one on the `vite-app-ts` can show the correct property type.

## Monorepo

Another issue I had is, for some reason it keep showing this message every time I opened [a (monorepo) project](https://github.com/feryardiant/admin-app) where the vue app placed on `website` directory.

![image](https://user-images.githubusercontent.com/508665/126081652-b6a53c46-333d-4f7b-9fd6-c75c007aa204.png)

The issue still remain even after I add `@vue/runtime-dom` as [`devDependencies`](https://github.com/feryardiant/admin-app/blob/master/website/package.json#L30) and in fact it is Vue 3 project, not Vue 2.

## So

- Could you please explain why is it possible? 
- For non-TS project, is it expected behaviour?
- Is it possible to make it behave exactly the same way as for TS?

---

### Info

OS :  Win 10 Home x64 10.0.19043
Node v14.17
PNPM v6.10
VS Code v1.58.2 (user setup)
Volar v0.26.4

Thank you in advance

**Comment by johnsoncodehk** [comment, marked as answer]
Hi @feryardiant, regarding the goals of this project, you can see https://github.com/johnsoncodehk/volar/discussions/168. You can understand it as close to the native TS development experience (capability and performance).

This 2 problems is also regressions, fixed in https://github.com/johnsoncodehk/volar/commit/f17c4d922bdf71648d514394fe3f2d82199f1ab0 (#319) and https://github.com/johnsoncodehk/volar/commit/8d2425054bde8e82d702fc6126e0b4418aafc013.

**Reply by feryardiant** [reply]
Awesome! Thank you
